web/src/app/theme/ThemePicker.tsx

```diff
@@ -98,15 +98,22 @@ export default function ThemePicker(): JSX.Element {
               <input
                 id='custom-color-picker'
                 onChange={(e) => {
-                  setSourceColor(e.target.value)
+                  if (e.target.value !== sourceColor) {
+                    setSourceColor(e.target.value)
+                  }
+                }}
+                onInput={(e) => {
+                  if (e.currentTarget.value !== sourceColor) {
+                    setSourceColor(e.currentTarget.value)
+                  }
```

Comment:
Can we trigger a `change` event in the test instead of binding to `onInput`? Also, why the need for the `!==` check?

Reply:
Through testing I found that I needed to have `onInput` as well in order to properly set the value. When I was attempting to trigger a change event nothing was happening even when I was setting force to true. I added the if statement just in case both a trigger and a change event were called simultaneously, mostly as a sanity check